### PR TITLE
new index in alert_log table

### PR DESCRIPTION
**Pull request by Negatifff**
at least speeding up the query to the alert_log table for the 
- 'alerts' widget in includes/html/table/alerts.inc.php : 111
- 'alertlog' widget includes/html/table/alertlog.inc.php : 87

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Negatifff** [comment]
trying to add this query
`create index `rule_and_device` on alert_log (`rule_id`,`device_id`);`
I hope I did everything right

**Comment by bofh80** [comment]
class AddDevicePerfIndex extends Migration
you want to change to that a new unique name like:
class AddAlertLogNewIndex extends Migration
￼(assuming that's not in use too i did not check :) )

**Comment by Negatifff** [comment]
Thanks, didn't notice

**Comment by bofh80** [comment]
i'm not 100% sure but maybe:
$table->index(['rule_id', 'device_id'], 'alert_log_rule_and_device_index');

But that's a *guess* .

**Comment by Negatifff** [comment]
> i'm not 100% sure but maybe:
> $table->index(['rule_id', 'device_id'], 'alert_log_rule_and_device_index');
> 
> But that's a _guess_ .

I did by analogy with database/migrations/2020_07_29_143221_add_device_perf_index.php

File: database/migrations/2020_09_22_172321_add_alert_log_index.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('alert_log', function (Blueprint $table) {
+            $table->index(['rule_id', 'device_id']);
+        });
+        Schema::table('alert_log', function (Blueprint $table) {
+            $table->index(['rule_id', 'device_id', 'state']);
+        });
```
Comment (Jellyfrog) [inline database/migrations/2020_09_22_172321_add_alert_log_index.php:21]:
```suggestion
        Schema::table('alert_log', function (Blueprint $table) {
            $table->index(['rule_id', 'device_id']);
            $table->index(['rule_id', 'device_id', 'state']);
        });
```

File: database/migrations/2020_09_22_172321_add_alert_log_index.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('alert_log', function (Blueprint $table) {
+            $table->dropIndex(['rule_id', 'device_id']);
+        });
+        Schema::table('alert_log', function (Blueprint $table) {
+            $table->dropIndex(['rule_id', 'device_id', 'state']);
+        });
```
Comment (Jellyfrog) [inline database/migrations/2020_09_22_172321_add_alert_log_index.php:36]:
```suggestion
        Schema::table('alert_log', function (Blueprint $table) {
            $table->dropIndex(['rule_id', 'device_id']);
            $table->dropIndex(['rule_id', 'device_id', 'state']);
        });
```

**Review by Jellyfrog — CHANGES_REQUESTED** [review]

File: database/migrations/2020_09_22_172321_add_alert_log_index.php (hunk 2 above)
Comment (Negatifff) [reply database/migrations/2020_09_22_172321_add_alert_log_index.php:36]: thanks

**Review by Jellyfrog — APPROVED** [review]

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-68-release-changelog-september-2020/13525/1

**Pull request merged**